src/VisualStudio/VisualStudioDiagnosticsToolWindow/Loggers/OutputWindowLogger.cs

```diff
@@ -41,97 +57,27 @@ public bool IsEnabled(FunctionId functionId)
-            {
-                s_instance.WriteLineInternal(value);
-            }
-
-            public OutputPane()
-            {
-                _serviceProvider = ServiceProvider.GlobalProvider;
-
-                var componentModel = (IComponentModel)_serviceProvider.GetService(typeof(SComponentModel));
-                _threadingContext = componentModel.GetService<IThreadingContext>();
-            }
-
-            private IVsOutputWindowPane _doNotAccessDirectlyOutputPane;
-
-            private void WriteLineInternal(string value)
-            {
-                var pane = GetPane();
-                if (pane == null)
-                {
-                    return;
-                }
-
-                pane.OutputStringThreadSafe(value + Environment.NewLine);
-            }
-
-            private IVsOutputWindowPane GetPane()
+            _threadingContext.JoinableTaskFactory.RunAsync(async () =>
             {
-                if (_doNotAccessDirectlyOutputPane == null)
-                {
-                    _threadingContext.JoinableTaskFactory.Run(async () =>
-                   {
-                       await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                       if (_doNotAccessDirectlyOutputPane != null)
-                       {
-                           // check whether other one already initialized output window.
-                           // the output API already handle double initialization, so this is just quick bail
-                           // rather than any functional issue
-                           return;
-                       }
-
-                       var outputWindow = (IVsOutputWindow)_serviceProvider.GetService(typeof(SVsOutputWindow));
-
-                       // this should bring outout window to the front
-                       _doNotAccessDirectlyOutputPane = CreateOutputPane(outputWindow);
-                   });
-                }
-
-                return _doNotAccessDirectlyOutputPane;
-            }
-
-            private IVsOutputWindowPane CreateOutputPane(IVsOutputWindow outputWindow)
-            {
-                _threadingContext.ThrowIfNotOnUIThread();
-
-                // Try to get the workspace pane if it has already been registered
-                var workspacePaneGuid = s_outputPaneGuid;
-
-                // If the pane has already been created, CreatePane returns it
-                if (ErrorHandler.Succeeded(outputWindow.CreatePane(ref workspacePaneGuid, "Roslyn Logger Output", fInitVisible: 1, fClearWithSolution: 1)) &&
-                    ErrorHandler.Succeeded(outputWindow.GetPane(ref workspacePaneGuid, out var pane)))
-                {
-                    return pane;
-                }
-
-                return null;
-            }
+                using var outputChannelStore = await _serviceBrokerClient.GetProxyAsync<IOutputChannelStore>(VisualStudioServices.VS2019_4.OutputChannelStore).ConfigureAwait(false);
```

Comment:
@AArnott I'm getting a COM exception when invoking this to get the proxy.  Wanted to check that I wasn't doing something obviously insane with regards to the service broker before checking the service impl

From the stack it looks possible it's having trouble loading the package providing the service, in which case I'll follow up with the owners of this particular service
```
"Error HRESULT E_FAIL has been returned from a call to a COM component."
"   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   
at Microsoft.VisualStudio.Services.VsTask.RethrowException(AggregateException e)\r\n   
at Microsoft.VisualStudio.Services.VsTask.InternalGetResult(Boolean ignoreUIThreadCheck)\r\n   
at Microsoft.VisualStudio.Services.VsTask.GetResult()\r\n   
at Microsoft.VisualStudio.Services.ServiceBroker.GlobalBrokeredServiceContainer.View.<LoadProfferingPackageAsync>d__19.MoveNext()\r\n   
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   
at Microsoft.VisualStudio.Services.ServiceBroker.GlobalBrokeredServiceContainer.View.<GetProxyAsync>d__17`1.MoveNext()\r\n   
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   
at System.Threading.Tasks.ValueTask`1.get_Result()\r\n   
at System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1.ConfiguredValueTaskAwaiter.GetResult()\r\n   
at Microsoft.ServiceHub.Framework.ServiceBrokerClient.<>c__DisplayClass21_0`1.<<GetProxyAsync>b__0>d.MoveNext()\r\n   
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   
at Microsoft.VisualStudio.Threading.AwaitExtensions.ExecuteContinuationSynchronouslyAwaiter`1.GetResult()\r\n  
 at Microsoft.VisualStudio.Threading.AsyncLazy`1.<>c__DisplayClass13_1.<<GetValueAsync>b__0>d.MoveNext()\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   
 at Microsoft.VisualStudio.Threading.JoinableTask.<JoinAsync>d__68.MoveNext()\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   
 at Microsoft.VisualStudio.Threading.JoinableTask`1.<JoinAsync>d__3.MoveNext()\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   
 at Microsoft.ServiceHub.Framework.ServiceBrokerClient.<GetProxyAsync>d__21`1.MoveNext()\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n  
  at System.Threading.Tasks.ValueTask`1.get_Result()\r\n   
  at System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1.ConfiguredValueTaskAwaiter.GetResult()\r\n   
  at Microsoft.CodeAnalysis.Internal.Log.OutputWindowLogger.<>c__DisplayClass10_0.<<WriteLine>b__0>d.MoveNext() in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\VisualStudio\\VisualStudioDiagnosticsToolWindow\\Loggers\\OutputWindowLogger.cs:line 78\r\n   
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   
  at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()\r\n   
  at Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run(Func`1 asyncMethod)\r\n   
  at Microsoft.CodeAnalysis.Internal.Log.OutputWindowLogger.WriteLine(String value) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\VisualStudio\\VisualStudioDiagnosticsToolWindow\\Loggers\\OutputWindowLogger.cs:line 76\r\n   
  at Microsoft.CodeAnalysis.Internal.Log.OutputWindowLogger.LogBlockStart(FunctionId functionId, LogMessage logMessage, Int32 uniquePairId, CancellationToken cancellationToken) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\VisualStudio\\VisualStudioDiagnosticsToolWindow\\Loggers\\OutputWindowLogger.cs:line 65\r\n   
  at Microsoft.CodeAnalysis.Internal.Log.AggregateLogger.LogBlockStart(FunctionId functionId, LogMessage logMessage, Int32 uniquePairId, CancellationToken cancellationToken) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\Workspaces\\Core\\Portable\\Log\\AggregateLogger.cs:line 141\r\n   at Microsoft.CodeAnalysis.Internal.Log.Logger.RoslynLogBlock.Construct(ILogger logger, FunctionId functionId, LogMessage logMessage, Int32 blockId, CancellationToken cancellationToken) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\Workspaces\\Core\\Portable\\Log\\Logger.LogBlock.cs:line 55\r\n   at Microsoft.CodeAnalysis.Internal.Log.Logger.CreateLogBlock(FunctionId functionId, LogMessage message, Int32 blockId, CancellationToken cancellationToken) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\Workspaces\\Core\\Portable\\Log\\Logger.LogBlock.cs:line 20\r\n   at Microsoft.CodeAnalysis.Internal.Log.Logger.LogBlock(FunctionId functionId, LogMessage logMessage, CancellationToken token) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\Workspaces\\Core\\Portable\\Log\\Logger.cs:line 340\r\n   at Microsoft.CodeAnalysis.Editor.Implementation.CommentSelection.ToggleLineCommentCommandHandler.<CollectEditsAsync>d__8.MoveNext() in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\EditorFeatures\\Core\\Implementation\\CommentSelection\\ToggleLineCommentCommandHandler.cs:line 65\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at Roslyn.Utilities.TaskExtensions.WaitAndGetResult_CanCallOnBackground[T](Task`1 task, CancellationToken cancellationToken) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\Workspaces\\Core\\Portable\\Utilities\\TaskExtensions.cs:line 87\r\n   at Roslyn.Utilities.TaskExtensions.WaitAndGetResult[T](Task`1 task, CancellationToken cancellationToken) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\Workspaces\\Core\\Portable\\Utilities\\TaskExtensions.cs:line 74\r\n   at Microsoft.CodeAnalysis.Editor.Implementation.CommentSelection.AbstractCommentSelectionBase`1.ExecuteCommand(ITextView textView, ITextBuffer subjectBuffer, TCommand command, CommandExecutionContext context) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\EditorFeatures\\Core\\Implementation\\CommentSelection\\AbstractCommentSelectionBase.cs:line 112\r\n   at Microsoft.CodeAnalysis.Editor.Implementation.CommentSelection.ToggleLineCommentCommandHandler.ExecuteCommand(ToggleLineCommentCommandArgs args, CommandExecutionContext context) in C:\\Users\\dabarbet\\source\\repos\\roslyn\\src\\EditorFeatures\\Core\\Implementation\\CommentSelection\\ToggleLineCommentCommandHandler.cs:line 54\r\n   at Microsoft.VisualStudio.Commanding.CommandHandlerExtensions.ExecuteCommand[T](ICommandHandler commandHandler, T args, Action nextCommandHandler, CommandExecutionContext executionContext)\r\n   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService.<>c__DisplayClass13_1`1.<Execute>b__1()\r\n   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call, Predicate`1 exceptionFilter)"
```

```
Managed Debugging Assistant 'FatalExecutionEngineError' 
  Message=Managed Debugging Assistant 'FatalExecutionEngineError' : 'The runtime has encountered a fatal error. The address of the error was at 0x07384298, on thread 0x73dc. The error code is 0x80131623. This error may be a bug in the CLR or in the unsafe or non-verifiable portions of user code. Common sources of this bug include user marshaling errors for COM-interop or PInvoke, which may corrupt the stack.'```

Reply:
I agree with your analysis: it looks like a package load bug in the proffering package.